Enterprise-CMCS/managed-care-review
# MCR-3839 restrict 3rd party API access by IP address

## Summary

- `configuration/allowed_ip_addresses` created in the parameter store
- a default IP address is supplied in `.envr` for local
- the `thirdPartyAuthn` function compares the current IP address with those in the parameter store value. If the IP address isn't allowed it will return an error and not allow the request to go through

#### Related issues
https://jiraent.cms.gov/browse/MCR-3839

#### Screenshots
<img width="1111" alt="Screenshot 2024-02-01 at 12 24 10 PM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/1d7649a9-d52d-4ae1-923c-0b84beac1663">


## QA guidance
 ```
curl -s https://t4rnz8nwfk.execute-api.us-east-1.amazonaws.com/mcr3839restrictaute0128/v1/graphql/external -X POST \
-H "Authorization: Bearer {jwt}" \
-H "Content-Type: application/json" \
--data '{"query":"query IndexRates {\n  indexRates {\n    totalCount\n    edges {\n      node {\n       id\n      }\n    }\n  }\n}"}'
```

The IP address that hits the review app changes daily. You can run the cURL command above and receive an error. Check the AWS logs for the outputted ip address that you can then add to the allowed_ip_addresses parameter store value. Once the correct IP address is added there you'll be able to hit the API
